src/instrumentation-client.ts

```
@@ -9,15 +14,23 @@ import {
 } from "./shared/constants/env.client";
 import { IS_DEV, IS_PROD } from "./shared/constants/env.flags";
 
+/**
+ * Escapes special characters in a string for use in a regular expression.
+ *
+ * @param {string} str - The raw string to escape.
+ * @returns {string} The escaped string safe for RegExp construction.
+ */
 function escapeRegExp(str: string) {
-  return str.replaceAll(/[$()*+.?[\\\]^{|}]/g, String.raw`\$&`);
+  return str.replaceAll(/[$()*+.?[\\\\\\]^{|}]/g, String.raw`\\$&`);
```

ESLint

[error] /[$()*+.?[\\\\\\]^{|}]/g can be optimized to /[$()*+.?[\\]^{|}]/g.

(unicorn/better-regex)